source/infrastructure/lib/api/model-schema/deploy-usecase-body.ts

```diff
@@ -58,6 +58,10 @@ export const deployUseCaseBodySchema: JsonSchema = {
             description: 'Deploy the CloudFront based UI for the use case',
             default: true
         },
+        ExistingCognitoUserPoolId: {
+            type: [JsonSchemaType.STRING, JsonSchemaType.NULL],
```

Comment:
If the intention is to use the default user pool created by the deployment dashboard when this is not provided, I would simply leave the type as `string`, and expect a user of the API to not pass any value for ExistingCognitoUserPoolId if they want to use the default.

Reply:
The problem occurs when we want to update the ExistingCognitoUserPoolId: 
(image user set the cognitouserpool id, but then wants to update it to the default one?)

Passing / not passing string doesn't indicate whether we want to explicitly reset user pool to the default one. With giving ability to explicitly set it as null, we can use this value when "Use Existing User Pool = false" in the UI.


let me know if this makes sense.

Reply:
Understood. That is a workflow I had not considered, and I will need to think about it more. We block certain types of edits (e.g. if you use an existing kendra index for RAG, an edit can't then tell the use case to provision a new kendra index), and I think this would be a potentially good candidate for that approach. 

As an alternative, perhaps if a user deploys with a custom user pool, perhaps they could return to the default one by editing and providing the id/client of the default pool in the wizard.